platform/commonUI/mobile/src/DeviceClassifier.js

```diff
@@ -44,10 +44,25 @@ define(
          */
         function MobileClassifier(agentService, $document) {
             var body = $document.find('body');
-            Object.keys(DeviceMatchers).forEach(function (key) {
+
+            Object.keys(DeviceMatchers).forEach(function (key, index, array) {
                 if (DeviceMatchers[key](agentService)) {
                     body.addClass(key);
                 }
+
+                if ((index === (array.length - 1)) && agentService.isMobile()) {
```

Comment:
Can you help me understand this logic? The way I read it the first clause will ensure that the code block inside the if statement is always invoked exactly once, and only if the user agent is a mobile device. Is this the intended logic? If so, why is it inside a for loop? I think I'm missing something here. Was the intention to actually invoke the `DeviceMatcher` logic for touch? If so, it would be better to just use the string literal 'touch'

Reply:
I think I was trying to solve this issue another way and forgot to remove the if block outside the for loop. There is no reason it should be inside. Thanks for the catch!